## 1. DATA COLLECTION

In [62]:
import pandas as pd

In [63]:
df=pd.read_csv("fake_or_real_news.csv")
df.head(10)

Unnamed: 0                                              title  \
0        8476                       You Can Smell Hillary’s Fear   
1       10294  Watch The Exact Moment Paul Ryan Committed Pol...   
2        3608        Kerry to go to Paris in gesture of sympathy   
3       10142  Bernie supporters on Twitter erupt in anger ag...   
4         875   The Battle of New York: Why This Primary Matters   
5        6903                                        Tehran, USA   
6        7341  Girl Horrified At What She Watches Boyfriend D...   
7          95                  ‘Britain’s Schindler’ Dies at 106   
8        4869  Fact check: Trump and Clinton at the 'commande...   
9        2909  Iran reportedly makes new push for uranium con...   

                                                text label  
0  Daniel Greenfield, a Shillman Journalism Fello...  FAKE  
1  Google Pinterest Digg Linkedin Reddit Stumbleu...  FAKE  
2  U.S. Secretary of State John F. Kerry said Mon...  REAL  
3  — Kaydee King (@KaydeeKing) November 9, 2016 T...  FAKE  
4  It's primary day in New York and front-runners...  REAL  
5    \nI’m not an immigrant, but my grandparents ...  FAKE  
6  Share This Baylee Luciani (left), Screenshot o...  FAKE  
7  A Czech stockbroker who saved more than 650 Je...  REAL  
8  Hillary Clinton and Donald Trump made some ina...  REAL  
9  Iranian negotiators reportedly have made a las...  REAL

In [64]:
df.shape

(6335, 4)

In [65]:
counts=df.label.value_counts()
counts

label
REAL    3171
FAKE    3164
Name: count, dtype: int64

## 2. Data PreProcessing

In [66]:
df.isnull().sum()

Unnamed: 0    0
title         0
text          0
label         0
dtype: int64

In [67]:
new_df=df.drop("title",axis="columns")

In [68]:
new_df.head()

Unnamed: 0                                               text label
0        8476  Daniel Greenfield, a Shillman Journalism Fello...  FAKE
1       10294  Google Pinterest Digg Linkedin Reddit Stumbleu...  FAKE
2        3608  U.S. Secretary of State John F. Kerry said Mon...  REAL
3       10142  — Kaydee King (@KaydeeKing) November 9, 2016 T...  FAKE
4         875  It's primary day in New York and front-runners...  REAL

## 3. EDA(Exploratory Data Analysis)

In [69]:
import plotly.express as px
import plotly.graph_objects as go
color=["Green","Red"]
fig=go.Figure(data=[go.Bar(x=["REAL","FAKE"],y=counts,text=counts,textposition="auto",marker_color=color)])
fig.show()


## 4. Model Selection and Training

In [70]:
X=df.text
y=df.label
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.2)


In [71]:
from sklearn.feature_extraction.text import TfidfVectorizer
tfidf=TfidfVectorizer(stop_words="english",max_df=0.7)


In [72]:
tfidf1=tfidf.fit_transform(X_train).toarray()
tfidf_test=tfidf.transform(X_test)

In [73]:
from sklearn.linear_model import PassiveAggressiveClassifier
model=PassiveAggressiveClassifier(max_iter=50)
model.fit(tfidf1,y_train)


PassiveAggressiveClassifier(max_iter=50)

In [74]:
model.score(tfidf_test,y_test)

0.9297553275453828

Getting the accuracy score of about 93% by using PassiveAggressiveClassifier